### ISSUE TITLE
sound (and music) outputs are routed to surround Rear Left and Rear Right, should be routed to Front Left / Front Right

### ISSUE THREAD

**Issue by lespin** [libgdx/libgdx#7468]
#### Issue details

When playing music and sounds, sound is routed to surround Rear Left and Rear Right. 

Every other program routes stereo to Front Left / Front Right, so that's what is expected.

#### Reproduction steps/code

Nothing special, just the standard

```
Music music = Gdx.audio.newMusic(Gdx.files.internal("mymusic.mp3"));
music.play();
```
`mymusic.mp3` beeing a stereo file.

#### Version of libGDX and/or relevant dependencies

gdxVersion=1.12.1

#### Stacktrace

No exception is raised

#### Please select the affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [ ] Windows
- [x] Linux
- [ ] macOS

**Comment by Frosty-J** [comment]
Could you try libGDX version 1.11.0? You could also try 1.12.2-SNAPSHOT, though I am not expecting it to help.

I've looked through the changes and the only one I can see that might be a cause is #6980.

**Comment by lespin** [comment]
Thanks for your answer. I did a quick check with **version 1.11.0** and **version 1.12.2-SNAPSHOT**.

**version 1.11.0**

Works perfectly fine. Client name is `alsoft`, output shows two channels (which is the expected behavior)

        $ pw-dump  | grep alsoft
        "client.name": "alsoft",
        "client.name": "alsoft",
        "node.description": "alsoft",
        "node.name": "alsoft",
        "port.alias": "alsoft:channel_1",
        "port.alias": "alsoft:channel_2",

**version 1.12.2-SNAPSHOT**

stereo output only routed to Rear (i.e. Exhibits the same issue as 1.12.1). Client name is `java`, output shows four channels.

        $ pw-dump  | grep java
        "application.name": "java",
        "application.process.binary": "java",
        "application.name": "java",
        "application.process.binary": "java",
        "node.description": "java",
        "node.name": "java",
        "object.path": "java:output_0",
        "port.alias": "java:output_FL",
        "object.path": "java:output_1",
        "port.alias": "java:output_FR",
        "object.path": "java:output_2",
        "port.alias": "java:output_RL",
        "object.path": "java:output_3",
        "port.alias": "java:output_RR",

**Comment by lespin** [comment]
Just a small addition: when playing music and sound with 1.12.1 ; the music is played on the front left and right channels, while the sounds are played on the rear left and right channels.

**Comment by Frosty-J** [comment]
I don't know why sounds only would be directed to the rear. If you were to run gdx-tests (download the repository, run `fechNatives` Gradle task, run `Lwjgl3TestStarter`), the tests of interest would be: `AudioChangeDeviceTest`, `AudioDeviceTest`, `MusicTest`, `SoundTest`.

Since the odds of finding someone else here who has a similar setup to sort it all out with are practically nil, alternative audio backends that might work better include https://github.com/Hangman/TuningFork and https://github.com/rednblackgames/gdx-miniaudio.

**Comment by SimonIT** [comment]
I also noticed this when I tried it on my 5.1 setup. I thought it was intentional. I have no real experience with OpenAL tho and not sure I can find some to dive into it.

**Comment by lespin** [comment]
@SimonIT  Yes, i'm sure we're not the only one having this problem. 4 channel soundcards or 5.1 setups are far from uncommon. 

I tried to trace what happen in a debug session but the libgdx / wjgl / openal layers makes very it hard to investigate without prior knowledge.

@Frosty-J I ran the tests you mentioned on latest git version :

`AudioChangeDeviceTest` -> plays on left front channel
`AudioDeviceTest` -> plays on left font + right front channel
`MusicTest` ->  plays on left font + right front channel
`SoundTest` -> plays on left rear + right rear channel

I'll have a look at MiniAudio as it's also multiplatform and seems to offer a lot more features, thanks a lot for the pointers.

**Comment by Hangman** [comment]
Small issue tracker overview of the three channel routing/mixing options in AL:

### AL_FALSE
All channels are seen as virtual and get routed to arbitry output channels. This is kinda predictable on standard stereo hardware though.
- Causes issues with mono speakers [https://github.com/libgdx/libgdx/issues/6631](https://github.com/libgdx/libgdx/issues/6631)
- Fixes the issue mentioned here

### AL_DROP_UNMATCHED_SOFT
Drops channels that don't match a direct output channel.
- Not an option, I guess

### AL_REMIX_UNMATCHED_SOFT
- Causes the issue mentioned here ([received the same bug report from a user of my game](https://steamcommunity.com/app/1338220/discussions/0/3722819643258905733/))
- Fixes [https://github.com/libgdx/libgdx/issues/6631](https://github.com/libgdx/libgdx/issues/6631)

In TuningFork, the channel virtualization method can be set on a per-source-basis, but I settled on AL_FALSE as the default (which users can also change).
There's no concept of a source in gdx right now, but providing users with the ability to change the used mixing option globally (probably in `Lwjgl3ApplicationConfiguration`) could be a first step to address this issue. It doesn't solve all problems, but I also don't see a reason against making this option available to the user.
